docs/pyproject.md

```diff
@@ -695,14 +695,25 @@ If you have multiple repositories configured, you can explicitly tell poetry whe
 requests = { version = "^2.13.0", source = "private" }
 ```
 
-{{% note %}}
-Be aware that declaring the python version for which your package
-is compatible is mandatory:
+You may also specify your project's compatible python versions in this section, instead of or in addition to `project.requires-python`.
 
 ```toml
 [tool.poetry.dependencies]
 python = "^3.7"
 ```
+
+{{% note %}}
+If you specify the compatible python versions in both `tool.poetry.dependencies` and in `project.requires-python`, then Poetry will use the information in `tool.poetry.dependencies` for locking, but the python versions must be a subset of those allowed by `project.requires-python`.
+For example, the following is invalid and will result in an error, because versions `4.0` and greater are allowed by `tool.poetry.dependencies`, but not by `project.requires-python`.
```

Comment:
```suggestion
If you specify the compatible python versions in both `tool.poetry.dependencies` and in `project.requires-python`, then Poetry will use the information in `tool.poetry.dependencies` for locking, but the python versions must be a subset of those allowed by `project.requires-python`.

For example, the following is invalid and will result in an error, because versions `4.0` and greater are allowed by `tool.poetry.dependencies`, but not by `project.requires-python`.
```

Minor nitpick.